src/go/k8s/internal/controller/redpanda/managed_decommission_controller.go

```diff
@@ -293,12 +294,13 @@ func podEvict(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alph
 func getPodFromRedpandaNodeID(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alpha1.Redpanda, nodeID int) (*corev1.Pod, error) {
 	log := l.WithName("getPodFromRedpandaNodeID")
 
-	pl, err := getPodList(ctx, c, rp.Namespace, labels.Set{
-		"app.kubernetes.io/component": componentLabelValue,
-		"app.kubernetes.io/instance":  rp.Name,
-	}.AsSelector())
+	selector, err := labels.Parse(fmt.Sprintf("!job-name,app.kubernetes.io/component=%s,app.kubernetes.io/instance=%s", componentLabelValue, rp.Name))
```

Comment:
Can we extract this label selector out into a helper function so it's less likely to get out of sync across all these usages?

FWIW, I'm pretty okay with panicking on errors from `labels.Parse` rather than handling them. If it panics once, it'll always panic and we can pretty easily verify the safety there of with unit tests. Just my 2 cents though!

Reply:
I moved to the dedicated function. I agree with the panic.